src/components/Checkbox/Checkbox.scss

```diff
@@ -1,82 +1,121 @@
+        // stylelint-disable-next-line selector-max-class, selector-max-specificity, max-nesting-depth
+        &::after {
+          background: color('white');
+        }
+      }
     }
-  }
 
-  &:checked,
-  &.Input-indeterminate {
-    // stylelint-disable-next-line selector-max-class
-    ~ .Icon {
-      transform: translate(-50%, -50%) scale(1);
+    &:checked,
+    &.Input-indeterminate {
+      // stylelint-disable-next-line selector-max-class, selector-max-specificity
+      ~ .Icon {
+        transform: translate(-50%, -50%) scale(1);
+      }
+    }
+
+    &:focus,
+    &:active {
+      // stylelint-disable-next-line selector-max-specificity
+      + .Backdrop {
+        @include control-backdrop(focused);
+      }
     }
-  }
 
-  &:focus,
-  &:active {
-    + .Backdrop {
-      @include control-backdrop(focused);
+    &:disabled {
+      // stylelint-disable-next-line selector-max-specificity
+      + .Backdrop {
+        @include control-backdrop(disabled);
+      }
+      // stylelint-disable-next-line selector-max-class, selector-max-specificity
+      ~ .Icon {
+        @include recolor-icon(color('ink', 'lightest'));
+      }
     }
   }
 
-  &:disabled {
-    + .Backdrop {
-      @include control-backdrop(disabled);
+  &.globalTheming {
+    &:focus {
+      // stylelint-disable-next-line selector-max-specificity
+      + .Backdrop {
+        @include focus-ring-animated($show: 1);
+      }
+    }
+    // stylelint-disable-next-line selector-max-specificity
+    &:active:not(:disabled),
+    &:checked,
+    &.Input-indeterminate {
+      // stylelint-disable-next-line selector-max-specificity
+      + .Backdrop {
+        @include control-backdrop-dl(active);
+      }
+      // stylelint-disable-next-line selector-max-specificity, selector-max-class
+      ~ .Icon {
+        transform: translate(-50%, -50%) scale(1);
+      }
     }
 
-    ~ .Icon {
-      @include recolor-icon(color('ink', 'lightest'));
+    &:disabled {
+      // stylelint-disable-next-line selector-max-specificity
+      + .Backdrop {
+        @include control-backdrop-dl(disabled);
+      }
+    }
+    // stylelint-disable-next-line selector-max-specificity
+    &:disabled:checked {
+      // stylelint-disable-next-line selector-max-specificity,
+      + .Backdrop {
+        background: var(--p-neutral-border-disabled);
+        // stylelint-disable-next-line selector-max-specificity, selector-max-class, max-nesting-depth, selector-max-combinators
+        &::before {
+          background: var(--p-neutral-border-disabled);
+        }
+      }
     }
   }
 }
 
 .Backdrop {
-  @include control-backdrop;
-  position: absolute;
-  top: 0;
-  bottom: 0;
-  left: 0;
-  right: 0;
-  pointer-events: none;
+  &:not(.globalTheming) {
+    position: absolute;
+    top: 0;
+    bottom: 0;
+    left: 0;
+    right: 0;
+    pointer-events: none;
+    @include control-backdrop;
+  }
+
+  &.globalTheming {
+    @include control-backdrop-dl;
```

Comment:
Is suffixing -dl an existing pattern already? Wondering if there's a better naming convention we could use or even start changing legacy mixin names so we're more clear of which ones should be avoided

Reply:
Changing our current mixins would be a breaking change. What do you think about putting inside

`/polaris-react/src/styles/_common.scss` which is part of our private API so that when we remove the `control-backdrop` mixin, we can simply drop the `-dl` and worry about breaking anyone?